### PR TITLE
isNil and isHexColorString

### PR DESCRIPTION
**Pull request by nkint**

### CODE REVIEW

File: packages/checks/src/is-hex-color-string.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { isString } from "util";
+
+export const isHexColorString = (x: any): x is string =>
+    isString(x) && /#([a-f0-9]{3}|[a-f0-9]{4}(?:[a-f0-9]{2}){0,2})\b/gi.test(x);
```
Comment (postspectacular) [inline packages/checks/src/is-hex-color-string.ts]: This regexp is actually not really watertight:

```js
/#([a-f0-9]{3}|[a-f0-9]{4}(?:[a-f0-9]{2}){0,2})\b/gi.test("hi #fff")
// true

/#([a-f0-9]{3}|[a-f0-9]{4}(?:[a-f0-9]{2}){0,2})\b/gi.test("#fff hi")
//true
```

Instead I'd propose using line start & end conditions to force the full string is an hex color:

```js
/^#([a-f0-9]{3}|[a-f0-9]{4}(?:[a-f0-9]{2}){0,2})$/i.test("hi #fff")
// false
```

Lastly, please pre-declare the regexp as module local const outside the function for a ~15-20% speed up... See [`isUUID`](https://github.com/thi-ng/umbrella/blob/master/packages/checks/src/is-uuid.ts) for ref

File: packages/checks/test/index.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@ import { isString } from "../src/is-string";
 import { isSymbol } from "../src/is-symbol";
 import { isTransferable } from "../src/is-transferable";
 import { isTypedArray } from "../src/is-typedarray";
+import { isNil } from "../src/is-nil";
+import { isHexColorString } from "../src/is-hex-color-string";
```
Comment (postspectacular) [inline packages/checks/test/index.ts]: Can you please add these 2 modules to the re-exports in [the package's `index.ts`](https://github.com/thi-ng/umbrella/blob/master/packages/checks/src/index.ts)

File: packages/checks/src/is-hex-color-string.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { isString } from "util";
```
Comment (postspectacular) [inline packages/checks/src/is-hex-color-string.ts]: This import is wrong, and should be:

```ts
import { isString } from "./is-string";
```

Btw. These wrong auto-imports are often biting me too in VSCode, especially with new source files. To remedy, I do the following:

- save the file
- choose "Restart TS server" from VS command pallette
- then add/quickfix missing imports

File: packages/checks/src/is-hex-color-string.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+
+export const isHexColorString = (x: any): x is string =>
```
[the remark this answers is not in the document]
Comment (nkint) [reply packages/checks/src/is-hex-color-string.ts]: 👍

File: packages/checks/src/is-hex-color-string.ts (hunk 1 above)
Comment (nkint) [reply packages/checks/src/is-hex-color-string.ts]: 👍

File: packages/checks/test/index.ts (hunk 2 above)
Comment (nkint) [reply packages/checks/test/index.ts]: 👍

File: packages/checks/src/is-hex-color-string.ts (hunk 3 above)
Comment (nkint) [reply packages/checks/src/is-hex-color-string.ts]: 👍

**Pull request merged**